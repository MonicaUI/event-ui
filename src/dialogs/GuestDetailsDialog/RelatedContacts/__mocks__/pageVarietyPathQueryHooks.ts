import { useQuery, QueryResult } from '@apollo/client';
import gql from 'graphql-tag';

export const gqlQuery = gql`
  query PageVarietyPath {
    event {
      registrationPath {
        guestRegistration {
          addGuestFromRelatedContacts {
            style {
              relatedContactsModalStyles
            }
            relatedContactsModalText
          }
        }
      }
    }
  }
`;

/**
 * Generates a custom hook to query registration path data for a specified page variety.
 */
export const useGuestRegistrationPageVarietyPathQuery = (): QueryResult => {
  const query = useQuery(gqlQuery);
  return query;
};
