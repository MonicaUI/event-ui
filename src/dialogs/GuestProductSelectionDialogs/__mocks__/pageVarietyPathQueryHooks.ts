import { useQuery, QueryResult } from '@apollo/client';
import gql from 'graphql-tag';

export const gqlQuery = gql`
  query PageVarietyPath {
    event {
      registrationPath {
        registration {
          admissionItems {
            display {
              fees
            }
          }
          sessions {
            display {
              fees
            }
          }
        }
      }
    }
  }
`;

/**
 * Generates a custom hook to query registration path data for a specified page variety.
 */
export const useRegistrationPageVarietyPathQuery = (): QueryResult => {
  const query = useQuery(gqlQuery);
  return query;
};
