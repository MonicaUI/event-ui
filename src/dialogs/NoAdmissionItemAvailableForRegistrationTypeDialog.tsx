import React from 'react';
import StandardDialog from './shared/StandardDialog';
import { openDialogContainer, closeDialogContainer } from 'nucleus-guestside-site/src/redux/modules/dialogContainer';
import DialogErrorStyles from './styles/DialogError.less';
import getDialogContainerStyle from './shared/getDialogContainerStyle';
import { withStyles } from './ThemedDialog';

const Dialog = withStyles(StandardDialog);

export const openNoAdmissionItemAvailableForRegistrationTypeDialog = (additionalActions?: $TSFixMe) => {
  return (dispatch: $TSFixMe, getState: $TSFixMe): $TSFixMe => {
    const translate = getState().text.translate;
    const boundCloseDialog = () => {
      dispatch(closeDialogContainer());
      if (additionalActions && typeof additionalActions === 'function') {
        dispatch(additionalActions());
      }
    };
    const dialog = (
      <Dialog
        onClose={boundCloseDialog}
        closeFallbackText={translate('EventGuestSide_StandardDialog_FallbackText_Close__resx')}
        message={translate('EventGuestSide_RegistrationTypeConflict_Title_resx')}
        subMessage={translate('EventGuestSide_RegistrationTypeError_NoAdmissionItemAvailableHelpText_resx')}
        title={translate('EventGuestSide_RegistrationTypeConflict_Title_resx')}
        icon="error"
        iconModifier="error"
        classes={DialogErrorStyles}
      />
    );
    return dispatch(
      openDialogContainer(dialog, boundCloseDialog, {
        classes: { dialogContainer: DialogErrorStyles.dialogContainer },
        style: { dragContainer: getDialogContainerStyle(getState()) }
      })
    );
  };
};
