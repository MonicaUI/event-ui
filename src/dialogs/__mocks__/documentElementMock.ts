export const getMockedMessageContainer = (): $TSFixMe => {
  // Mock a DOM to play around
  document.body.innerHTML = `
    <div>
    <div name="messageContainer" tabIndex=-1>message container</div>
    </div>
  `;
  const mockMessageContainerFocus = jest.fn();
  const mockMessageContainer = document.getElementsByName('messageContainer')[0];
  mockMessageContainer.focus = mockMessageContainerFocus;
  return mockMessageContainer;
};

export const getMockedFieldInputs = (fieldNames: $TSFixMe): $TSFixMe => {
  // Mock a DOM to play around
  document.body.innerHTML = `
    <div>
    ${fieldNames.map(fieldName => `<input name=${fieldName}>input-${fieldName}</input>`)}
    </div>
  `;
  const mockInputFocus = jest.fn();
  const mockInputContainers = [];
  fieldNames.forEach(fieldName => {
    const mockInputContainer = document.getElementsByName(fieldName)[0];
    mockInputContainer.focus = mockInputFocus;
    mockInputContainers.push(mockInputContainer);
  });
  return mockInputContainers;
};
