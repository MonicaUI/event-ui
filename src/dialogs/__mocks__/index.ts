/* eslint-env jest */
import noopAction from '../../testUtils/noopAction';

export const openAlreadyRegisteredDialog = jest.fn(noopAction);

export const openDeclineRegistrationDialog = jest.fn(noopAction);

export const openStartNewRegistrationDialogFromPageLanding = jest.fn(noopAction);

export const openStartNewRegistrationDialogDuringRegistration = jest.fn(noopAction);

export const startNewRegistrationAndNavigateToRegistration = jest.fn(noopAction);

export const openEventWaitlistDialog = jest.fn(noopAction);

export const openEventTemporaryClosedErrorDialog = jest.fn(noopAction);

export const openNoAdmissionItemAvailableForRegistrationTypeDialog = jest.fn(noopAction);

export const openEventStatusDialog = jest.fn(noopAction);

export const openPrivateEventErrorDialog = jest.fn(noopAction);

export const openRegistrationInOtherTabDialog = jest.fn(noopAction);

export const openKnownErrorDialog = jest.fn(noopAction);

export const openPaymentProcessingErrorDialog = jest.fn(noopAction);

export const openInformationAlreadyUsedDialog = jest.fn(noopAction);

export const openPartialRegistrationConfirmationDialog = jest.fn(noopAction);

export const openPreviewModeWarningDialog = jest.fn(noopAction);

export const openCapacityReachedDialog = jest.fn(noopAction);

export const openGroupCancelRegistrationDialog = jest.fn(noopAction);

export const openSessionOverlapWarningDialog = jest.fn(noopAction);

export const openGroupMemberRemoveDialog = jest.fn(noopAction);

export const openSharePromptDialog = jest.fn(noopAction);

export const openConfirmationDialog = jest.fn(noopAction);

export const openGuestProductSelectionDialog = jest.fn(noopAction);

export const openGroupRegistrationTypeDialog = jest.fn(noopAction);

export const openGuestDetailsDialog = jest.fn(noopAction);

export const openGuestRemoveDialog = jest.fn(noopAction);

export const openContactPlannerDialog = jest.fn(noopAction);

export const openCancelRegistrationSuccessConfirmationDialog = jest.fn(noopAction);

export const openCancelRegistrationDialog = jest.fn(noopAction);

export const openShoulderDateApprovalWarningDialog = jest.fn(noopAction);

export const openTransactionInProcessingErrorDialog = jest.fn(noopAction);
