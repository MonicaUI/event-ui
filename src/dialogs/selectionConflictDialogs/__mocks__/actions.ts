export const returnToProcessStart = jest.fn(() => () => {});
