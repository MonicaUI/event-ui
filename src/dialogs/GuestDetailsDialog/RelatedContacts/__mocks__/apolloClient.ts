import { MockedResponse } from '@apollo/client/testing';
import { gqlQuery } from './pageVarietyPathQueryHooks';

type RelatedContactsDialogMockData = {
  relatedContactsModalStyles: unknown;
  relatedContactsModalText: unknown;
};

export const getApolloClientMocks = (
  state: RelatedContactsDialogMockData
): MockedResponse<Record<string, $TSFixMe>>[] => {
  const { relatedContactsModalStyles = {}, relatedContactsModalText = {} } = state;
  return [
    {
      request: {
        query: gqlQuery
      },
      result: {
        data: {
          event: {
            registrationPath: {
              guestRegistration: {
                addGuestFromRelatedContacts: {
                  style: {
                    relatedContactsModalStyles
                  },
                  relatedContactsModalText
                }
              }
            }
          }
        }
      }
    }
  ];
};
