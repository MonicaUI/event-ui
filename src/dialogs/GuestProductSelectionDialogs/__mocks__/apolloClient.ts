import { MockedResponse } from '@apollo/client/testing';
import { gqlQuery } from './pageVarietyPathQueryHooks';

type GuestProductSelectionDialogMockData = {
  displayAdmissionItemsFees: unknown;
  displaySessionsFees: unknown;
};

export const getApolloClientMocks = (
  state: GuestProductSelectionDialogMockData
): MockedResponse<Record<string, $TSFixMe>>[] => {
  const { displayAdmissionItemsFees = {}, displaySessionsFees = {} } = state;
  return [
    {
      request: {
        query: gqlQuery
      },
      result: {
        data: {
          event: {
            registrationPath: {
              registration: {
                admissionItems: {
                  display: {
                    fees: displayAdmissionItemsFees
                  }
                },
                sessions: {
                  display: {
                    fees: displaySessionsFees
                  }
                }
              }
            }
          }
        }
      }
    }
  ];
};
