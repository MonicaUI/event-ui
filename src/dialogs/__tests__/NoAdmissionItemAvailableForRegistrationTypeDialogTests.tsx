import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import DialogContainer from 'nucleus-guestside-site/src/containers/DialogContainer';
import createStoreWithMiddleware from 'nucleus-guestside-site/src/redux/devCreateStoreWithMiddleware';
import { combineReducers } from 'redux';
import { openNoAdmissionItemAvailableForRegistrationTypeDialog } from '../NoAdmissionItemAvailableForRegistrationTypeDialog';
import React from 'react';
import dialogContainer from 'nucleus-guestside-site/src/redux/modules/dialogContainer';
import { wait } from '../../testUtils';
import EventSnapshot from '../../../fixtures/EventSnapshot.json';

// eslint-disable-next-line jest/no-mocks-import
import { getMockedMessageContainer } from '../__mocks__/documentElementMock';
getMockedMessageContainer();

jest.mock('nucleus-core/containers/Transition');

const store = createStoreWithMiddleware(
  combineReducers({
    dialogContainer,
    text: (x = {}) => x,
    website: (x = {}) => x
  }),
  {
    text: {
      translate: (resx, tokens) => (tokens ? `${resx}:${JSON.stringify(tokens)}` : resx)
    },
    website: {
      theme: EventSnapshot.eventSnapshot.siteEditor.website.theme
    }
  }
);

const additionalFunctionalityCallBack = jest.fn(() => () => {});

describe('NoAdmissionItemAvailableForRegistrationTypeDialog', () => {
  const dialog = mount(
    <Provider store={store}>
      <DialogContainer spinnerMessage="spinnerMessage" message="message" />
    </Provider>
  );

  test('matches snapshot when opened', () => {
    store.dispatch(openNoAdmissionItemAvailableForRegistrationTypeDialog());
    expect(dialog).toMatchSnapshot();
  });
  test('make sure close handler is set up', async () => {
    dialog.update();
    dialog.find('[data-cvent-id="close"]').hostNodes().simulate('click');
    await wait(0);
    expect(dialog).toMatchSnapshot();
  });
});

describe('NoAdmissionItemAvailableForRegistrationTypeDialog additional functionality on close', () => {
  const dialog = mount(
    <Provider store={store}>
      <DialogContainer spinnerMessage="spinnerMessage" message="message" />
    </Provider>
  );

  test('make sure close handler is set up and additional functionality is called', async () => {
    store.dispatch(openNoAdmissionItemAvailableForRegistrationTypeDialog(additionalFunctionalityCallBack));
    dialog.update();
    expect(dialog).toMatchSnapshot();
    dialog.find('[data-cvent-id="close"]').hostNodes().simulate('click');
    await wait(0);
    expect(dialog).toMatchSnapshot();
    expect(additionalFunctionalityCallBack).toHaveBeenCalled();
  });
});
