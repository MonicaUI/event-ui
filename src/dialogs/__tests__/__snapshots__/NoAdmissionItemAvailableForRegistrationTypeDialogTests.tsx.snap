// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoAdmissionItemAvailableForRegistrationTypeDialog additional functionality on close make sure close handler is set up and additional functionality is called 1`] = `
Array [
  <div
    class=""
  >
    <div
      class=""
    />
    <span
      class="mockNucleusCoreTransition"
    >
      <div
        class=""
      >
        <div
          aria-label="Dialog"
          role="dialog"
          style="outline: none;"
          tabindex="0"
        >
          <div
            class=""
            style="background-color: rgb(243, 59, 21); border-radius: 100px; position: absolute; left: 512px; top: 384px;"
          >
            <div
              class=""
            />
            <div
              class=""
            >
              <div
                class="dialogContainer"
              >
                <div
                  class="dialogChildren"
                >
                  <div>
                    <div
                      aria-describedby="dialogHeader"
                      class=""
                      style="background-color: rgb(221, 226, 230); border-radius: 0;"
                    >
                      <div
                        class=""
                        id="dialogHeader"
                        style="background-color: rgb(249, 32, 242); padding: 15px 1.5625rem 15px 1.5625rem;"
                      >
                        <h2
                          class=""
                          style="color: rgb(255, 255, 255); font-family: Comic Sans MS, Comic Sans, Chalkboard, Marker Felt, sans-serif; font-size: 3.5625rem; font-style: normal; font-weight: 700; line-height: 1.3;"
                        >
                          <div
                            class=""
                            data-cvent-id="close"
                            role="button"
                            tabindex="0"
                          >
                            <span
                              class="container"
                            >
                              <span
                                aria-label="EventGuestSide_StandardDialog_FallbackText_Close__resx"
                                class=""
                                role="img"
                              />
                            </span>
                          </div>
                          EventGuestSide_RegistrationTypeConflict_Title_resx
                        </h2>
                      </div>
                    </div>
                    <div
                      class=""
                      name="messageContainer"
                      style="background-color: rgb(243, 59, 21); padding: 0px 1.25rem 1.875rem 1.25rem;"
                      tabindex="-1"
                    >
                      <span
                        class="container"
                      >
                        <span
                          aria-hidden="true"
                          class=""
                          role="img"
                        />
                      </span>
                      <div
                        class=""
                        style="font-family: PT Sans, sans-serif; font-weight: 400; font-size: 2.5rem; color: rgb(39, 63, 105); line-height: 1.3; font-style: normal; background-color: transparent; border-radius: 0; padding: 0px 0px 0px 0px;"
                      >
                        EventGuestSide_RegistrationTypeConflict_Title_resx
                      </div>
                      <p
                        class=""
                      >
                        EventGuestSide_RegistrationTypeError_NoAdmissionItemAvailableHelpText_resx
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>,
]
`;

exports[`NoAdmissionItemAvailableForRegistrationTypeDialog additional functionality on close make sure close handler is set up and additional functionality is called 2`] = `
Array [
  <div
    class=""
  >
    <div
      class=""
    />
    <span
      class="mockNucleusCoreTransition"
    />
  </div>,
]
`;

exports[`NoAdmissionItemAvailableForRegistrationTypeDialog make sure close handler is set up 1`] = `
Array [
  <div
    class=""
  >
    <div
      class=""
    />
    <span
      class="mockNucleusCoreTransition"
    />
  </div>,
]
`;

exports[`NoAdmissionItemAvailableForRegistrationTypeDialog matches snapshot when opened 1`] = `
Array [
  <div
    class=""
  >
    <div
      class=""
    />
    <span
      class="mockNucleusCoreTransition"
    >
      <div
        class=""
      >
        <div
          aria-label="Dialog"
          role="dialog"
          style="outline: none;"
          tabindex="0"
        >
          <div
            class=""
            style="background-color: rgb(243, 59, 21); border-radius: 100px; position: absolute; left: 512px; top: 384px;"
          >
            <div
              class=""
            />
            <div
              class=""
            >
              <div
                class="dialogContainer"
              >
                <div
                  class="dialogChildren"
                >
                  <div>
                    <div
                      aria-describedby="dialogHeader"
                      class=""
                      style="background-color: rgb(221, 226, 230); border-radius: 0;"
                    >
                      <div
                        class=""
                        id="dialogHeader"
                        style="background-color: rgb(249, 32, 242); padding: 15px 1.5625rem 15px 1.5625rem;"
                      >
                        <h2
                          class=""
                          style="color: rgb(255, 255, 255); font-family: Comic Sans MS, Comic Sans, Chalkboard, Marker Felt, sans-serif; font-size: 3.5625rem; font-style: normal; font-weight: 700; line-height: 1.3;"
                        >
                          <div
                            class=""
                            data-cvent-id="close"
                            role="button"
                            tabindex="0"
                          >
                            <span
                              class="container"
                            >
                              <span
                                aria-label="EventGuestSide_StandardDialog_FallbackText_Close__resx"
                                class=""
                                role="img"
                              />
                            </span>
                          </div>
                          EventGuestSide_RegistrationTypeConflict_Title_resx
                        </h2>
                      </div>
                    </div>
                    <div
                      class=""
                      name="messageContainer"
                      style="background-color: rgb(243, 59, 21); padding: 0px 1.25rem 1.875rem 1.25rem;"
                      tabindex="-1"
                    >
                      <span
                        class="container"
                      >
                        <span
                          aria-hidden="true"
                          class=""
                          role="img"
                        />
                      </span>
                      <div
                        class=""
                        style="font-family: PT Sans, sans-serif; font-weight: 400; font-size: 2.5rem; color: rgb(39, 63, 105); line-height: 1.3; font-style: normal; background-color: transparent; border-radius: 0; padding: 0px 0px 0px 0px;"
                      >
                        EventGuestSide_RegistrationTypeConflict_Title_resx
                      </div>
                      <p
                        class=""
                      >
                        EventGuestSide_RegistrationTypeError_NoAdmissionItemAvailableHelpText_resx
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>,
]
`;
